### PR TITLE
Add tox

### PR DESCRIPTION
**Pull request by alvaro-garcia-carrasco**
Add tox as virtual environment manager.

[Tox](https://tox.readthedocs.io/en/latest/) is a virtual environment manager. It is very useful to test python packages under different environments. It also makes the integration with CI platforms easier. Tox is used by nice projects like Requests and Numpy.

**Why do we need tox?**
It makes testing much easier. With tox we can recreate locally the testing on the CI environment.

How to run it:
`$ pip install tox`
`$ tox`

If you use Conda:
`$ pip install tox-conda`
`$ tox`

You can select on which environment to run tox with either of the following commands:
`$ tox -e [ENV]`

Tox caches downloaded dependencies, if you change those, use:
`$ tox --recreate`

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,31 +1,23 @@
 language: python
 dist: xenial  # required for python 3.7+
-python:
-  - "3.5"
-  - "3.6"
-  - "3.7"
-  - "3.8-dev"
-cache: pip
```
Comment (agrinh) [inline .travis.yml]: Can the pip cache be kept?

File: .travis.yml [hunk 2]
```diff
@@ -1,31 +1,23 @@
...
-  - isort --check-only
-  - isort --diff
+install:
+  - pip install tox
 
-  # Check for unused symbols/imports and PEP8 formatting.
-  - flake8 --version
-  - flake8 --disable-noqa
+script:
+  - tox
```
Comment (agrinh) [inline .travis.yml]: Does tox install sidekick?

Comment (alvaro-garcia-carrasco) [reply .travis.yml]: yes

File: .travis.yml (hunk 1 above)
Comment (alvaro-garcia-carrasco) [reply .travis.yml]: Good catch! I have added it back now

**Comment by agrinh** [comment]
@alvaro-garcia-carrasco Please make sure to merge master into this branch

**Comment by alvaro-garcia-carrasco** [comment]
Continue in #33

**Pull request closed without merging**